ZuluContent/Engines/Craft/Core/Repair.cs

```diff
@@ -1,371 +1,135 @@
-					}
-					else
-					{
-						if ( CheckWeaken( from, skill, clothing.HitPoints, clothing.MaxHitPoints ) )
-						{
-							clothing.MaxHitPoints -= toWeaken;
-							clothing.HitPoints = Math.Max( 0, clothing.HitPoints - toWeaken );
-						}
-
-						if ( CheckRepairDifficulty( from, skill, clothing.HitPoints, clothing.MaxHitPoints ) )
-						{
-							number = 1044279; // You repair the item.
-							m_CraftSystem.PlayCraftEffect( from );
-							clothing.HitPoints = clothing.MaxHitPoints;
-						}
-						else
-						{
-							number = usingDeed ? 1061137 : 1044280; // You fail to repair the item. [And the contract is destroyed]
-							m_CraftSystem.PlayCraftEffect( from );
-						}
-
-						toDelete = true;
-					}
-				}
-				else if( !usingDeed && targeted is BlankScroll )
-				{
-					SkillName skill = m_CraftSystem.MainSkill;
-
-					if( from.Skills[skill].Value >= 50.0 )
-					{
-						((BlankScroll)targeted).Consume( 1 );
-						RepairDeed deed = new RepairDeed( RepairDeed.GetTypeFor( m_CraftSystem ), from.Skills[skill].Value, from );
-						from.AddToBackpack( deed );
-
-						number = 500442; // You create the item and put it in your backpack.
-					}
-					else
-						number = 1047005; // You must be at least apprentice level to create a repair service contract.
-				}
-				else if ( targeted is Item )
-				{
-					number = usingDeed? 1061136 : 1044277; // That item cannot be repaired. // You cannot repair that item with this type of repair contract.
-				}
-				else
-				{
-					number = 500426; // You can't repair that.
-				}
-
-				if( !usingDeed )
-				{
-					CraftContext context = m_CraftSystem.GetContext( from );
-					from.SendGump( new CraftGump( from, m_CraftSystem, m_Tool, number ) );
-				}
-				else
-				{
-					from.SendLocalizedMessage( number );
-
-					if( toDelete )
-						m_Deed.Delete();
-				}
-			}
-		}
-	}
+    {
+        public Repair()
+        {
+        }
+
+        public static void Do(Mobile from, CraftSystem craftSystem, BaseTool tool)
+        {
+            from.Target = new InternalTarget(craftSystem, tool);
+            from.SendLocalizedMessage(1044276); // Target an item to repair.
+        }
+
+        private class InternalTarget : Target
+        {
+            private CraftSystem m_CraftSystem;
+            private BaseTool m_Tool;
+
+            public InternalTarget(CraftSystem craftSystem, BaseTool tool) : base(2, false, TargetFlags.None)
+            {
+                m_CraftSystem = craftSystem;
+                m_Tool = tool;
+            }
+
+            private void DamageItem(Mobile from, int damageAmount, IRepairable repairable, ref int number)
+            {
+                damageAmount = Math.Max(damageAmount, 1);
+
+                if (damageAmount >= repairable.HitPoints)
+                {
+                    number = 500424; // You destroyed the item.
+                    from.SendFailureMessage("You fail and destroy the item.");
```

Comment:
Might be some Cliloc's that cover these messages, e.g. `500424: "You destroyed the item."`

Reply:
Cool thanks! Done :)